packages/js/src/dashboard/routes/alert-center.js

```diff
@@ -1,18 +1,33 @@
+import { select } from "@wordpress/data";
 import { __ } from "@wordpress/i18n";
 import { Paper, Title } from "@yoast/ui-library";
 import { PremiumUpsellList } from "../../shared-admin/components/premium-upsell-list";
+import SidebarRecommendations from "../../shared-admin/components/sidebar-recommendations";
 import { Notifications, Problems } from "../components";
-import SidebarRecommendations from "../components/sidebar-recommendations";
 import { useSelectDashboard } from "../hooks";
+import { STORE_NAME } from ".././constants";
+
+/**
+ * Uses the store's selector to get whether a promotion is active.
+ *
+ * @param {string} promotionID The promotion ID.
+ *
+ * @returns {bool} Whether the promotion is active.
+ */
+const isPromotionActive = ( promotionID ) => {
+	return select( STORE_NAME ).isPromotionActive( promotionID );
+};
```

Comment:
I _think_ you could also do inside the render:
```js
const { isPromotionActive } = useSelect( STORE_NAME );
```

That is meant for when you don't need to re-render on any change. And this function should never change.
Here is some blog about using that for event callbacks: https://developer.wordpress.org/news/2024/03/28/how-to-work-effectively-with-the-useselect-hook/#but-call-them-outside-when-you-re-in-an-event-handler

If deciding to change, please also tackle the other ones

Reply:
Decided to move forward with this 👍